src/EventLoop.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Interop\Async\EventLoop;
+
+final class EventLoop
+{
+    /**
+     * @var EventLoopDriver
+     */
+    private static $driver = null;
+
+    /**
+     * Execute a callback within the scope of an event loop driver.
+     *
+     * @param callable $callback The callback to execute
+     * @param EventLoopDriver $driver The event loop driver
+     *
+     * @return void
+     */
+    public static function execute(callable $callback, EventLoopDriver $driver)
+    {
+        $previousDriver = self::$driver;
+
+        self::$driver = $driver;
+
+        try {
+            $callback();
+
+            self::$driver->run();
+        } finally {
+            self::$driver = $previousDriver;
+        }
+    }
+
+    /**
+     * Retrieve the event loop driver that is in scope.
+     * 
+     * @return EventLoopDriver
+     */
+    public static function get()
+    {
+        if (null === self::$driver) {
+            throw new \RuntimeException('Not within the scope of an event loop driver');
+        }
+
+        return self::$driver;
+    }
+
+    /**
+     * Stop the event loop.
+     * 
+     * @return void
+     */
+    public static function stop()
+    {
+        self::get()->stop();
```

Comment:
Agree; else you have no possibility to intentionally run destructors etc.
Especially important for tests.